pkg/pillar/cmd/zedrouter/patchenvelopes.go

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) 2023 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package zedrouter
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+// PatchEnvelopes is a structure representing
+// Patch Envelopes exposed to App instances via metadata server
+// for more info check docs/PATCH-ENVELOPES.md
+// Must be created by calling NewPatchEnvelopes()
+//
+// Internally, PatchEnvelopes structure stores envelopes which
+// come from EdgeDevConfig parsed by zedagent. This envelopes contains
+// both inline binary artifacts which are ready to be downloaded by app instances
+// and volume references, which are handled by volumemgr.
+// So PatchEnvelopes struct has completedVolumes and contentTreeStatus to store
+// information of all volumes and contentTree handled by volumemgr to link them with
+// patch envelope volume references. ContentTreeStatus is used to retrieve SHA of underlying
+// file.
+// PatchEnvelopes are used by app instances via Get() which returns list of available PatchEnvelopeInfo
```

Comment:
Nit, but shouldn't the Get() be "http GET"?

Reply:
No, because Get() is PatchEnvelope method, which is called inside metadata server handler

Reply:
Ah - that was far from clear because the "used by app instances" sounds like what the app instance invokes and not what the metadata server invokes.

Reply:
Updated doc :)